src/pages/home/report/ReactionList/PopoverReactionList/BasePopoverReactionList.js

```diff
@@ -1,245 +0,0 @@
-        }
-
-        // Hide the list when all reactions are removed
-        const isEmptyList = !_.some(lodashGet(this.props.emojiReactions, [this.props.emojiName, 'users']));
-        if (!isEmptyList) {
-            return;
-        }
-
-        this.hideReactionList();
-    }
-
-    componentWillUnmount() {
-        if (!this.dimensionsEventListener) {
-            return;
-        }
-        this.dimensionsEventListener.remove();
-    }
-
-    /**
-     * Get the BasePopoverReactionList anchor position
-     * We calculate the achor coordinates from measureInWindow async method
-     *
-     * @returns {Promise<Object>}
-     */
-    getReactionListMeasuredLocation() {
-        return new Promise((resolve) => {
-            if (this.reactionListAnchor.current) {
-                this.reactionListAnchor.current.measureInWindow((x, y) => resolve({x, y}));
-            } else {
-                resolve({x: 0, y: 0});
-            }
-        });
-    }
-
-    /**
-     * Get the reaction information.
-     *
-     * @param {Object} selectedReaction
-     * @param {String} emojiName
-     * @returns {Object}
-     */
-    getReactionInformation(selectedReaction, emojiName) {
-        if (!selectedReaction) {
-            return {
-                emojiName: '',
-                reactionCount: 0,
-                emojiCodes: [],
-                hasUserReacted: false,
-                users: [],
-            };
-        }
-
-        const {emojiCodes, reactionCount, hasUserReacted, userAccountIDs} = EmojiUtils.getEmojiReactionDetails(emojiName, selectedReaction, this.props.currentUserPersonalDetails.accountID);
-
-        const users = PersonalDetailsUtils.getPersonalDetailsByIDs(userAccountIDs, this.props.currentUserPersonalDetails.accountID, true);
-        return {
-            emojiName,
-            emojiCodes,
-            reactionCount,
-            hasUserReacted,
-            users,
-        };
-    }
-
-    /**
-     * Show the ReactionList modal popover.
-     *
-     * @param {Object} [event] - A press event.
-     * @param {Element} reactionListAnchor - reactionListAnchor
-     */
-    showReactionList(event, reactionListAnchor) {
-        const nativeEvent = event.nativeEvent || {};
-        this.reactionListAnchor.current = reactionListAnchor;
-        this.getReactionListMeasuredLocation().then(({x, y}) => {
-            this.setState({
-                cursorRelativePosition: {
-                    horizontal: nativeEvent.pageX - x,
-                    vertical: nativeEvent.pageY - y,
-                },
-                popoverAnchorPosition: {
-                    horizontal: nativeEvent.pageX,
-                    vertical: nativeEvent.pageY,
-                },
-                isPopoverVisible: true,
-            });
-        });
-    }
-
-    /**
-     * This gets called on Dimensions change to find the anchor coordinates for the action BasePopoverReactionList.
-     */
-    measureReactionListPosition() {
```

Comment:
You removed a lot of methods in this file, these weren't used at all or what is the reason?

Reply:
Ahh you moved it to a hook, nice. Did you test it?

Reply:
@blazejkustra Thanks!
It wasn't working properly but I just pushed a commit that it does. The functionality is exactly the same as before.